go.mod

```diff
@@ -1,5 +1,8 @@
 module github.com/snapcore/snapd
 
+// TODO: update to 1.18, doing this naively breaks the builds because it
+// seems this enables automatic vendoring, see
```

Comment:
which builds are broken? all of them? also the snap build?

Reply:
All debian builds which means by implication the snap build (as it builds the debs first):
```
go: inconsistent vendoring in /home/gopath/src/github.com/snapcore/snapd/_build/src/github.com/snapcore/snapd:
	github.com/canonical/go-tpm2@v0.0.0-20210827151749-f80ff5afff61: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
...
```
I am looking into how to fix this but it's tricky :/

Reply:
as a short term fix to unblock us, can't we just rewrite the go.mod with s/go 1.18/go 1.13/ at the start of the debian builds? or too hacky? I know we do various hacks there already

Reply:
It seems the culprit is the missing `vendor/modules.txt` file and the error is just misleading (i.e. the file is not there at all).
So something like:
```
diff --git a/packaging/ubuntu-16.04/rules b/packaging/ubuntu-16.04/rules
index 9415092ef0..074d53a2c7 100755
--- a/packaging/ubuntu-16.04/rules
+++ b/packaging/ubuntu-16.04/rules
@@ -176,8 +176,9 @@ override_dh_clean:
 override_dh_auto_build:
        # usually done via `go generate` but that is not supported on powerpc
        ./mkversion.sh
+       # generate dh-golang does not copy modules.txt
+       cp -a vendor/modules.txt _build/src/$(DH_GOPKG)/vendor
        # ensure auto-generated version is also in the build-tree
-       # XXX: do we need this?
        (cd _build/src/$(DH_GOPKG)/ && ../../../../../mkversion.sh)
        # Build golang bits
        mkdir -p _build/src/$(DH_GOPKG)/cmd/snap/test-data
```
and including vendor/modules.txt might be needed but I would prefer this as a followup as it may uncover more dragons and the existing PR archives the goal of moving to go1.18

Reply:
I opened a test PR with this in https://github.com/snapcore/snapd/pull/12719

Reply:
ah ok